okapi-core/src/main/java/org/folio/okapi/managers/ModuleManager.java

```diff
@@ -115,15 +115,31 @@ public ModuleManager(ModuleStore moduleStore, boolean local) {
           return Future.failedFuture(new OkapiError(ErrorType.USER, res));
         }
       }
-      return createList2(newList);
+      return createList2(newList).compose(x -> {
+        if (deleteObsolete) {
+          return deleteObsolete();
+        } else {
+          return Future.succeededFuture();
+        }
+      });
     });
   }
 
+  Future<Void> deleteObsolete() {
+    return modules.getAll()
+        .compose(ares -> {
+          List<ModuleDescriptor> newList = new LinkedList<>(ares.values());
+          Future<Void> future = Future.succeededFuture();
+          for (ModuleDescriptor md: ModuleUtil.getObsolete(newList)) {
```

Comment:
How can we ensure that a md is not deleted when the module is currently deployed?

Reply:
done now